pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -161,48 +163,46 @@ def _give_curves_from_samples(
         ]
 
 
-def gather_data(seq: pulser.sequence.Sequence, gather_output: bool) -> dict:
+def gather_data(sampled_seq: SequenceSamples) -> dict:
     """Collects the whole sequence data for plotting.
 
     Args:
         seq: The input sequence of operations on a device.
-        gather_output: Whether to gather the modulated output curves.
 
     Returns:
         The data to plot.
     """
     # The minimum time axis length is 100 ns
-    total_duration = max(
-        seq.get_duration(include_fall_time=gather_output), 100
```

Comment:
The issue in the documentation comes from here: the samples have to be extended if `gather_output` is set to True. 
- `gather_output` should be always False in `draw_samples`.
- `gather_output` is equal to `draw_modulation` in `draw_sequence` (because samples have to be extended if `draw_modulation` is set to True, because they will be modulated afterwards)

Reply:
Better idea: You should add a `shown_duration` option, that represents the duration shown on the plots. You should call `extend_duration` with this `shown duration`. 
- In `draw_samples`, `shown_duration` is None
- In `draw_sequence`, `shawn_duration` is `seq.get_duration(include_fall_time=draw_modulation)`

Reply:
I tried to do this (last commit), but the problem seems to remain

Reply:
It's now solved ;)